From d939a0bcc690b0761acc2247d4e8827f7eb149b9 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Wed, 20 Mar 2024 10:32:34 +0100
Subject: [PATCH 0709/1695] dts: stm32: fix wake-up from Standby for stm32mp15
 dhcor boards

Fix wake-up from Standby for stm32mp15 dhcor boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 97d6d9139395fddde67c2511402dab7455c2f538 dts: stm32: fix wake-up from Standby for stm32mp15 dhcor boards]
Change-Id: Ic91fa46ab5b983a776e14c71d3b3205a63f7f810
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/367963
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi   | 2 ++
 core/arch/arm/dts/stm32mp15xx-dhcor-io1v8.dtsi | 1 +
 core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi   | 1 +
 3 files changed, 4 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi b/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi
index e9428e5b5..75969e5d8 100644
--- a/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi
@@ -78,6 +78,7 @@
 		gpio = <&gpiog 3 GPIO_ACTIVE_LOW>;
 		regulator-always-on;
 		regulator-boot-on;
+		st,regulator-warm-boot-on;
 		vin-supply = <&vdd>;
 	};
 };
@@ -349,6 +350,7 @@
 				regulator-min-microvolt = <2900000>;
 				regulator-max-microvolt = <2900000>;
 				regulator-boot-on;
+				st,regulator-warm-boot-on;
 				interrupts = <IT_CURLIM_LDO5 0>;
 			};
 
diff --git a/core/arch/arm/dts/stm32mp15xx-dhcor-io1v8.dtsi b/core/arch/arm/dts/stm32mp15xx-dhcor-io1v8.dtsi
index 75172314d..3ebb4577c 100644
--- a/core/arch/arm/dts/stm32mp15xx-dhcor-io1v8.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dhcor-io1v8.dtsi
@@ -14,6 +14,7 @@
 		regulator-max-microvolt = <1800000>;
 		regulator-always-on;
 		regulator-boot-on;
+		st,regulator-warm-boot-on;
 		vin-supply = <&vdd>;
 	};
 };
diff --git a/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi b/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi
index b3d0aa324..afd549c74 100644
--- a/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi
@@ -175,6 +175,7 @@
 				regulator-max-microvolt = <2900000>;
 				interrupts = <IT_CURLIM_LDO5 0>;
 				regulator-boot-on;
+				st,regulator-warm-boot-on;
 			};
 
 			v1v8: ldo6 {
-- 
2.47.2

